### PR TITLE
[RFM] Update/streamline CI builds (drop py3.5, add py3.8, remove/update old pins/workarounds)

### PR DESCRIPTION
**Pull request by gojomo**
PR to test changes to build recipes - adjusting Python & other-library versions. See discussion #2713.

(Replaces #2714)

### CODE REVIEW

**Comment by gojomo** [comment]
Removing older explicit pins/wheel-installations-from-nonstandard-places left everything working (and in fact resolved py3.6 errors in my other #2698 work-in-progress).

Adding py3.8 worked fine on Travis-CI (linux) builds, but is currently triggering a mysterious failure on Appveyor (windows). 

(Specifically: At a certain point where the Appveyor-py3.7 build works fine downloading needed packages, the Appveyor 3.8 build seems to want to try compiling Cython – successfully – and then cythonizing Numpy – unsuccessfully – itself. It tries this even if an explicit wheels-based install of Cython can be demonstrated to have worked, a few steps earlier, so there's some other strange condition, not clear in logs, causing the Appveyor-py3.8 process to not use the same pre-built resources as the Appveyor-py3.7 process does.)

**Comment by gojomo** [comment]
This re-based PR now does 3 things:

* a more complete, less-buggy replacement of deprecated `time.clock()` than prior #2730 

* remove Python 3.5 from buld specs, adding {ython 3.8 for Travis-CI (& commented-out specs for Python 3.8 on Appveyor, for whenever the [issues mentioned above](https://github.com/RaRe-Technologies/gensim/pull/2715#issuecomment-569457589) resolve)

*  adds some automatic `gdb`-driven core-dump debugging (on Travis_CI), based on what I found helpful while working on (still-in-progress) #2698 – so that at any future time any of our tests core-dump on the Travis-CI system, we'll have a head-start in seeing exactly where

With all tests passing, I think it's ready for review/merge.

File: appveyor.yml [hunk 1]
```diff
@@ -50,9 +52,15 @@ install:
   - "powershell ./continuous_integration/appveyor/install.ps1"
   - "SET PATH=%PYTHON%;%PYTHON%\\Scripts;%PATH%"
   - "python -m pip install -U pip tox"
-  - "pip --version"
 
-  # Check that we have the expected version and architecture for Python
+  # next line only to demo that py3.8-on-Appveyor *could* install Cython from wheel just fine,
```
Comment (piskvorky) [inline appveyor.yml]: Adding capitalization/punctuation to make it easier to see where one statement ends and another begins.

```suggestion
  # Next line only to demo that py3.8-on-Appveyor *could* install Cython from wheel just fine,
```

File: appveyor.yml [hunk 2]
```diff
@@ -50,9 +52,15 @@ install:
...
+  # despite mysterious following attempt/failure to build-and-use Cython on that one Appveyor config
+  # delete when py3.8-on-Appveyor starts working normally
```
Comment (piskvorky) [inline appveyor.yml]:
```suggestion
  # Delete when py3.8-on-Appveyor starts working normally,
```

Comment (piskvorky) [inline appveyor.yml]:
```suggestion
  # delete when py3.8-on-Appveyor starts working normally.
```

File: appveyor.yml [hunk 3]
```diff
@@ -50,9 +52,15 @@ install:
   - "powershell ./continuous_integration/appveyor/install.ps1"
   - "SET PATH=%PYTHON%;%PYTHON%\\Scripts;%PATH%"
   - "python -m pip install -U pip tox"
-  - "pip --version"
 
-  # Check that we have the expected version and architecture for Python
+  # next line only to demo that py3.8-on-Appveyor *could* install Cython from wheel just fine,
+  # despite mysterious following attempt/failure to build-and-use Cython on that one Appveyor config
```
Comment (piskvorky) [inline appveyor.yml]:
```suggestion
  # despite mysterious following attempt/failure to build-and-use Cython on that one Appveyor config.
```

**Comment by piskvorky** [comment]
Whoa, please wait for @mpenkov's review before merging.

**Comment by gojomo** [comment]
Sorry, thought one "approved" was enough. (What is the preferred standard for approved-to-merge?)

Do you want me to use the github 'revert' button (which creates a new reversion PR & leaves add-then-revert in history), force-push a rollback, or just wait to see if his post-review turns up anything?

(I'd prefer to do the latter, as I've already rebased my other PR-in-progress on `develop` with these changes and it minimizes history complications, & I don't expect more than minor adjustments to this PR.)

**Comment by piskvorky** [comment]
Yes, "wait to see", after @mpenkov review. No need to revert.

**Comment by ecederstrand** [comment]
Any ETA on a PyPI release that has packages for Python 3.8? Just wondering whether I should attempt a source install from a checkout of master into my Python 3.8 installation, or be patient :-)

**Comment by mpenkov** [comment]
This PR is unrelated to Python 3.8 compatibility with gensim. The PR only adds tests for our continuous integration testing environment.

If you look in PyPI, wheels for Py3.8 are already available:

https://pypi.org/project/gensim/#files

**Comment by ecederstrand** [comment]
I don't see wheels for Python 3.8 at that link? Python 3.7 is the newest I see binary wheels for.

Anyway, I managed to install from source, so this is not urgent for me anymore.

**Comment by mpenkov** [comment]
Sorry, you're right. I double-checked and 3.8 is missing. I've added a ticket here: https://github.com/MacPython/gensim-wheels/issues/22

**Pull request merged**